config/rollup.base.js

```diff
@@ -48,6 +49,8 @@ export const getBuildMetadata = ({ name, version }) => {
     packageVersion: version,
     buildDate: now.toLocaleString('en-US', { dateStyle: 'long' }),
     buildTime: now.toLocaleString('en-US', { timeStyle: 'long' }),
+    gitCommit: execSync('git rev-parse HEAD').toString().trim(),
+    gitBranch: execSync('git rev-parse --abbrev-ref HEAD').toString().trim(),
```

Comment:
I assume this is a temp solution - using isomorphic git seems safer

Reply:
isomorphic git is safer -- and it can have reasonable defaults if `git` isn't installed or if the exec fails

Reply:
> I assume this is a temp solution - using isomorphic git seems safer

In Node.js, `execSync` is a method provided by the `child_process` module that allows for synchronous execution of shell commands from within a JavaScript program. This can be useful in certain scenarios, but it also comes with its own set of advantages and disadvantages when compared to a pure JavaScript implementation.

Executing Git commands using the `exec` function from Node.js's `child_process` module and using the `Isomorphic-git` library can both help you to interact with Git repositories from a JavaScript application. Here are some pros and cons I can suggest:

Using `execAsync('git ...')`

Pros:

Familiarity: If you're already used to Git command-line operations, using execAsync('git ...') allows you to apply your existing knowledge directly, using the same commands as you would in the terminal.

Full Git Features: The git command-line interface provides a comprehensive set of Git features. By invoking Git commands directly, you have access to this complete functionality.

Cons:

Handling Asynchronicity: The execAsync method is asynchronous, which might make our code more complex, particularly if you have a series of commands that must be executed in a specific order.

Error Handling: Processing errors can be more challenging with `execAsync` as we might need to parse standard error outputs or manage non-zero exit codes.

System Dependency: our code requires that git be installed on the system. This dependency might limit the environments where the code can run and could cause issues if different environments have varying versions of git.

Using `Isomorphic-git`

Pros:

Independence from System Git: `Isomorphic-git` is a pure JavaScript implementation and does not require git to be installed on the system. This improves the portability of our code.

Improved Control and Error Handling: Using `Isomorphic-git`, you can exert more control over the Git operations, and managing errors can be more robust and straightforward compared to interpreting command-line outputs.

Cons:

Learning Curve: Users may need to spend some time learning the `Isomorphic-git` API, even if you are already familiar with Git command-line operations.

Potential Limitations in Functionality: While `Isomorphic-git `is comprehensive, it may not support every Git feature. 

In our case I prefer to use `execAsync('git ...')` because of the simplicity of use and that we already require the installation of `git` CLI in our project

> isomorphic git is safer -- and it can have reasonable defaults if git isn't installed or if the exec fails

installing `git` is a pre perquisite of our development environment, users may encounter other problems before running this script in case they forgot to install git.

here are example of using both, I prefere the `execSync` option because of it's simplicity, and because in out case `git` is already a pre requisite, while `Isomorphic-git`  will requiere to install a new (big) pakcage.

```js
   try {
        const commitHash = execSync('git rev-parse HEAD').toString().trim();
        console.log(commitHash);
    } catch (err) {
        console.error('Failed to retrieve commit hash:', err);
    }
```

```js
   try {
        const commitHash = await git.resolveRef({ fs, dir: '.', ref: 'HEAD' });
        console.log(commitHash);
    } catch (err) {
        console.error('Failed to retrieve commit hash:', err);
    }
```

both options assume that they are being run from within a Git repository. If they aren't, both will fail with an error message.

IMHO the `execSync`is easier to read and understand while not requiring a third party package.

Can you give more arguments to support the safety concern ?

Reply:
> Can you give more arguments to support the safety concern ?
>

1. `execSync` is launching a default shell. Shells differ between systems which is source of various problems.
2.  the result is text and needs to be parsed from stdio

Security reasons do not apply in our current setup.